### PR TITLE
Predibase LLM Integration

### PR DESCRIPTION
**Pull request by Abhay-765**
# Predibase LLM Integration

This PR adds a Predibase integration as a custom LLM. I've also added an example notebook for reference.

Please let me know if there's anything I can do to improve the PR. When it's merged, feel free to tag https://twitter.com/predibase and https://twitter.com/AbhayyM as contributors.

### CODE REVIEW

File: llama_index/llms/predibase.py [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+            num_output=self.max_new_tokens,
+            model_name=self.model_name,
+        )
+
+    # "CompletionResponse"
+    def complete(self, prompt: str, **kwargs: Any) -> "CompletionResponse":
+        model_kwargs = {
+            "temperature": self.temperature,
+            "max_new_tokens": self.max_new_tokens,
+        }
```
Comment (Disiok) [inline llama_index/llms/predibase.py:59]: curious: does predibase LLM support any other kwargs?

So far we allow user to inject any LLM specific kwargs via the complete endpoint. Just wondering if there's additional configurability that's not yet exposed via the constructor args.

File: llama_index/llms/predibase.py [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+    ) -> None:
+        self.predibase_api_key = (
+            predibase_api_key
+            if predibase_api_key
+            else os.environ.get("PREDIBASE_API_TOKEN")
+        )
+        self.model_name = model_name
+        self.max_new_tokens = max_new_tokens
+        self.temperature = temperature
+        self.context_window = context_window
```
Comment (Disiok) [inline llama_index/llms/predibase.py:28]: curious: is there a way to automatically get the context window from your LLM implementations?

**Review by Disiok — APPROVED** [review]

File: llama_index/llms/predibase.py (hunk 1 above)
Comment (Abhay-765) [reply llama_index/llms/predibase.py:59]: Yep I saw that. These are the primary arguments users can currently control, though we are looking to add additional configurability in the future.

File: llama_index/llms/predibase.py (hunk 2 above)
Comment (Abhay-765) [reply llama_index/llms/predibase.py:28]: Not currently, though that is something we're considered adding

**Comment by Disiok** [comment]
Great, thanks for address the comments! 
Looks like there's some minor lint issue, could you do a last
``` 
make format;make lint
```
quicky and I can merge this after.

File: examples/predibase/Predibase Integration Demo.ipynb
Comment (Disiok) [inline examples/predibase/Predibase Integration Demo.ipynb:1]: I think some of the notes at the top of the notebook is from Vellum integration. Could you update this?

Comment (Abhay-765) [reply examples/predibase/Predibase Integration Demo.ipynb:1]: Updated!

**Pull request merged**